src/main/java/com/nutomic/syncthingandroid/syncthing/GetTask.java

```diff
@@ -55,9 +56,12 @@ protected String doInBackground(String... params) {
         String fullUri = params[0] + params[1];
         Log.v(TAG, "Calling Rest API at " + fullUri);
 
-        if (params.length == 5) {
+        if (params.length >= 5) {
             LinkedList<NameValuePair> urlParams = new LinkedList<>();
-            urlParams.add(new BasicNameValuePair(params[3], params[4]));
+            int paramCounter = 3;
+            while (paramCounter + 1 < params.length) {
+                urlParams.add(new BasicNameValuePair(params[paramCounter++], params[paramCounter++]));
+            }
```

Comment:
You could do `for (int paramCounter = 3; paramCounter + 1 < params.length; )` here.